samples/matter/common/cmake/source_common.cmake

```diff
@@ -41,7 +41,7 @@ if(CONFIG_CHIP_NUS)
     target_sources(app PRIVATE ${MATTER_COMMONS_SRC_DIR}/bt_nus/bt_nus_service.cpp)
 endif()
 
-if(CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL)
+if(CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL AND CONFIG_NCS_SAMPLE_MATTER_SETTINGS_STORAGE_BACKEND)
```

Comment:
I'm not sure if we want to create such dependency. Someone may want to use settings shell with our persistent storage module disabled and e.g. monitor usage of settings in Matter stack.

Reply:
Fair.